### ISSUE TITLE
[Bug] Imgui failed to display on Mac

### ISSUE THREAD

**Issue by Morcki** [taichi-dev/taichi#6481]
**Describe the bug**
Trace the bug report #6451 , which report that the latest release version of taichi (v1.2.0) failed to display imgui when using GGUI.

**To Reproduce**
Here is a simple sample code to reproduce this error.

```py
# sample code here
import taichi as ti

ti.init(arch=ti.vulkan, offline_cache=False)

pos = ti.Vector.field(3, dtype=ti.float32, shape=3)

display_mode = 0 
def show_options():
    global display_mode
    window.GUI.begin("Display Panel", 0.05, 0.1, 0.2, 0.15)
    display_mode = window.GUI.slider_int("display_mode", display_mode, 0, 1)
    window.GUI.end()


window = ti.ui.Window("Display Mesh", (1024, 1024), vsync=True)
canvas = window.get_canvas()
scene = ti.ui.Scene()
while window.running:
    # if comment the below line, it will get the correct result
    scene.particles(pos, radius=0.01, color=(1.0, 0.0, 0.0))
    canvas.scene(scene)

    show_options()
    window.show()
```

**Log/Screenshots**
The incorrect screenshots: 
<img width="396" alt="image" src="https://user-images.githubusercontent.com/34432001/198975798-0644b6ed-bf9d-44e6-85a8-dfe7a2ee900a.png">
The correct screenshots: 
<img width="298" alt="image" src="https://user-images.githubusercontent.com/34432001/198976340-41e382d5-9ed8-45a2-a025-24b5ef0c9f3b.png">

**Comment by ailzhang** [comment]
@Morcki should v1.2.1 wait for this fix? Thanks!

**Comment by Morcki** [comment]
> @Morcki should v1.2.1 wait for this fix? Thanks!

Very sorry for not noticing your comment, my apologize 🥲

The reason for this is because of the newest molten-vk version(1.2) while older version(1.1) doesn't have this problem.
I think we'd better fix it before v1.2.1 for we don't know how much users use the newest version of molten-vk.
